v2/internal/runner/github.go

```diff
@@ -0,0 +1,174 @@
+		return
+	}
+	owner = strings.Split(reponame, "/")[0]
+	repo = strings.Split(reponame, "/")[1]
+	return
+}
+
+// returns *github.Repository if passed github repo name
+func getGithubRepo(gitHubClient *github.Client, repoOwner, repoName string) (*github.Repository, error) {
+	var retried bool
+getRepo:
+	repo, _, err := gitHubClient.Repositories.Get(context.Background(), repoOwner, repoName)
+	if err != nil {
+		// retry with authentication
+		if gitHubClient = getGHClientWithToken(); gitHubClient != nil && !retried {
+			retried = true
+			goto getRepo
+		}
+		return nil, err
+	}
+	if repo == nil {
+		return nil, errors.Errorf("problem getting repository: %s/%s", repoOwner, repoName)
+	}
+	return repo, nil
+}
+
+// download the git repo to given path
+func (ctr *customTemplateRepo) cloneRepo(clonePath, githubToken string) error {
+	r, err := git.PlainClone(clonePath, false, &git.CloneOptions{
+		URL:  ctr.gitCloneURL,
+		Auth: getAuth(ctr.owner, githubToken),
+	})
+	if err != nil {
+		return errors.Errorf("%s/%s: %s", ctr.owner, ctr.reponame, err.Error())
+	}
+	// Add the user as well in the config. By default user is not set
+	config, _ := r.Storer.Config()
+	config.User.Name = ctr.owner
+	return r.SetConfig(config)
+}
+
+// performs the git pull on given repo
+func (ctr *customTemplateRepo) pullChanges(repoPath, githubToken string) error {
+	r, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return err
+	}
+	w, err := r.Worktree()
+	if err != nil {
+		return err
+	}
+	err = w.Pull(&git.PullOptions{RemoteName: "origin", Auth: getAuth(ctr.owner, githubToken)})
+	if err != nil {
+		return errors.Errorf("%s/%s: %s", ctr.owner, ctr.reponame, err.Error())
+	}
+	return nil
+}
+
+// getLocalRepoClonePath returns the clone path.
+// if same name repo directory exists from another owner then it appends the owner then and returns the path
+// eg. for nuclei-templates directory exists for projectdiscovery owner, then for ehsandeep/nuclei-templates it will return nuclei-templates-ehsandeep
+func (ctr *customTemplateRepo) getLocalRepoClonePath(downloadPath string) string {
+	if isDirectory(filepath.Join(downloadPath, ctr.reponame)) && !ctr.isRepoDirExists(filepath.Join(downloadPath, ctr.reponame)) {
+		return filepath.Join(downloadPath, ctr.reponame+"-"+ctr.owner)
+	}
+	return filepath.Join(downloadPath, ctr.reponame)
+}
+
+// isRepoDirExists take the path and checks if the same repo or not
+func (ctr *customTemplateRepo) isRepoDirExists(repoPath string) bool {
+	r, _ := git.PlainOpen(repoPath)
+	local, _ := r.Config()
+	return local.User.Name == ctr.owner // repo already cloned no need to rename and clone
+}
+
+// returns the auth object with username and github token as password
+func getAuth(username, password string) *http.BasicAuth {
+	if username != "" && password != "" {
+		return &http.BasicAuth{Username: username, Password: password}
+	}
+	return nil
+}
+
+// isDirectory returns true if path is directory
+func isDirectory(path string) bool {
```

Comment:
please check if `fileutil.FolderExists` already covers this check

Reply:
Updated as per your suggestion